dali/operators/image/remap/warp_affine_params.h

```diff
@@ -109,37 +115,53 @@ class WarpAffineParamProvider
     }
   }
 
-  void UseInputAsParams(const TensorList<CPUBackend> &input) {
+  void UseInputAsParams(const TensorList<CPUBackend> &input, bool invert) {
     CheckParamInput(input);
 
-    params_cpu_.data = static_cast<const MappingParams *>(input.raw_data());
-    params_cpu_.shape = { num_samples_ };
+    if (invert) {
+      auto *params = this->AllocParams(kernels::AllocType::Host);
+      for (int i = 0; i < num_samples_; i++) {
+        if (invert)
+          params[i] = static_cast<const MappingParams *>(input.raw_tensor(i))->invert();
```

Comment:
don't you have to assign params to params_cpu_
```
params_cpu_ = {params, {num_samples, }};
```
?

Reply:
This is the magic of `AllocParams` - called with `kernels::AllocType::Host` returns params_cpu_ .

Reply:
AllocParams does it